src/frontend/screens/LeaveProject/LeaveProject.tsx

```diff
@@ -0,0 +1,196 @@
+import React, { useContext, useState } from "react";
+import { defineMessage, FormattedMessage } from "react-intl";
+import { View, Text, StyleSheet, Image } from "react-native";
+import CheckBox from "@react-native-community/checkbox";
+import ConfigContext from "../../context/ConfigContext";
+import Button from "../../sharedComponents/Button";
+import { useNavigation } from "react-navigation-hooks";
+import {
+  NavigationStackOptions,
+  NavigationStackScreenComponent,
+} from "react-navigation-stack";
+import HeaderTitle from "../../sharedComponents/HeaderTitle";
+
+const m = defineMessage({
+  leaveProjectTitle: {
+    id: "screens.LeaveProject.LeaveProject.leaveProjectTitle",
+    defaultMessage: "Leave Project{projectName}?",
+  },
+  headerTitle: {
+    id: "screens.LeaveProject.LeaveProject.headerTitle",
+    defaultMessage: "Leave Project",
+  },
+  willDelete: {
+    id: "screens.LeaveProject.LeaveProject.willDelete",
+    defaultMessage:
+      "This will delete {numOfObservations} observations and {numOfPics} Pictures",
+  },
+  agreeToDelete: {
+    id: "screens.LeaveProject.LeaveProject.agreeToDelete",
+    defaultMessage: "I understand I will be deleting all data from my device.",
+  },
+  leaveButton: {
+    id: "screens.LeaveProject.LeaveProject.leaveButton",
+    defaultMessage: "Leave Project",
+  },
+  cancel: {
+    id: "screens.LeaveProject.LeaveProject.cancel",
+    defaultMessage: "Cancel",
+  },
+  confirmDelete: {
+    id: "screens.LeaveProject.LeaveProject.confirmDelete",
+    defaultMessage: "Please check the box to confirm",
+  },
+  syncWarning: {
+    id: "screens.LeaveProject.LeaveProject.syncWarning",
+    defaultMessage:
+      "Sync with a team member so you don't loose all your observations.",
+  },
+});
+
+const navOptions: NavigationStackOptions = {
+  headerTitle: () => (
+    <HeaderTitle style={{ color: "#ffffff" }}>
+      <FormattedMessage {...m.headerTitle} />
+    </HeaderTitle>
+  ),
+};
+
+export const LeaveProjectScreen: NavigationStackScreenComponent = () => {
+  const [confirmDelete, setConfirmDelete] = React.useState(false);
+  const [pristine, setPristine] = useState(true);
```

Comment:
i'd advocate for using a clearer name here such as `touched` or `checkboxTouched` (with the proper initial value of course)